internal/build/lifecycle_execution.go

```diff
@@ -135,11 +135,16 @@ func (l *LifecycleExecution) Run(ctx context.Context, phaseFactoryCreator PhaseF
 		}
 		buildCache = cache.NewImageCache(cacheImage, l.docker)
 	} else {
-		buildCache = cache.NewVolumeCache(l.opts.Image, l.opts.Cache.Build, "build", l.docker)
+		switch l.opts.Cache.Build.Format {
+		case cache.CacheVolume:
+			buildCache = cache.NewVolumeCache(l.opts.Image, l.opts.Cache.Build, "build", l.docker)
+		case cache.CacheBind:
+			buildCache = cache.NewBindCache(l.opts.Cache.Build, l.docker)
+		}
 	}
 
 	l.logger.Debugf("Using build cache volume %s", style.Symbol(buildCache.Name()))
-	if l.opts.ClearCache {
+	if l.opts.ClearCache && l.opts.Cache.Build.Format != cache.CacheBind {
```

Comment:
I have not allowed clear cache support for bind caches due to security reasons, let me know if this needs to be changed.

Reply:
If the user provides a very specific directory to use as a cache, why would we want to make an exception in this case but not in a named volume? I think the unexpected difference in behavior is more problematic than deleting files within a targeted directory.

Reply:
> If the user provides a very specific directory to use as a cache, why would we want to make an exception in this case but not in a named volume?

I feel like if user does something like `--cache="type=build;format=bind;name=."` alongwith `--clear-cache`, then we'd be cleaning up the current dir. How do we tackle this?

Reply:
That's an interesting callout. I would agree that it _could_ be dangerous. What if we completely restricted the combination of options instead of bypassing the functionality? IOW, what do you think about failing early if the user provides any `bind` format with `--clear-cache`.

We can provide a helpful message like "'--clear-cache' is not supported when using 'bind' cache. You may instead delete the cache directory manually.".

The only drawback is that it would prevent --clear-cache when using a mix of any other cache format.

---

Alternatively, what if we had a nested directory with an explicit name for bind? That way at least it's less likely that we could delete unexpected content. 

```
.
└── <user-defined>/
    └── <type>-cache/
```

For example, when the user provides:
`--cache="type=build;format=bind;name=./my-cache/"`

we create the parent directory (if missing), but additionally create a child directory `<type>-cache` (ie. `build-cache`) such as:

```
.
└── my-cache/
    └── build-cache/
```

then when the users provides the same flag `--cache="type=build;format=bind;name=./my-cache/"` plus `--clear-cache` we only delete the `my-cache/build-cache` directory.

WDYT?